Assets/MRTK/Core/Utilities/WebRequestRest/Response.cs

```diff
@@ -16,12 +16,15 @@ public struct Response
         /// <summary>
         /// Response body from the resource.
         /// </summary>
-        public string ResponseBody { get; }
+        public string ResponseBody => cachedResponseBody ?? (cachedResponseBody = responseBodyAction?.Invoke());
+        private string cachedResponseBody;
+        private System.Func<string> responseBodyAction;
 
         /// <summary>
         /// Response data from the resource.
         /// </summary>
-        public byte[] ResponseData { get; }
+        public byte[] ResponseData => responseDataAction?.Invoke();
```

Comment:
Should this be cached too?

Reply:
jup, just did it.
Now the cache is read in the case the constructor with concrete values is used and in the other case it's written to by accessing the callback first if there is one.

Reply:
Prefect, looks good to me!